server/api/CandyRoutes.js

```diff
@@ -0,0 +1,19 @@
+const router = require('express').Router()
+const SingleCandy = require('../db/models/Candy')
+
+router.get('/:id', async (req, res, next) => {
+  try {
+    console.log('HELLO I AM HERE ----------')
+    const selectedCandy = await SingleCandy.findByPk(req.params.id)
+    if (selectedCandy) {
+      res.json(selectedCandy)
+    } else {
+      res.status(404).send('Candy not found')
```

Comment:
I think you can use the "next"  method so that 404 handling happens in one place only.